js/component/dns-client/dns-packet.js

```diff
@@ -70,98 +63,99 @@ exports.getQuery = function(domain, query) {
   return u8;
 };
 
-var POINTER_VALUE = 0xc0;
-function isPointer(value) {
-  return (value & POINTER_VALUE) === POINTER_VALUE;
-}
+const POINTER_VALUE = 0xc0;
+const isPointer = value => (value & POINTER_VALUE) === POINTER_VALUE;
 
-function readHostname(reader) {
-  var labels = [];
+const readHostname = (reader) => {
+  let labels = [];
 
-  for (var z = reader.getOffset(); z < reader.len; ++z) {
-    var len = reader.readUint8();
-    if (0 === len) {
-      break;
-    }
+  for (let z = reader.getOffset(); z < reader.len; ++z) {
+    const len = reader.readUint8();
+    if (len === 0) break;
 
     if (isPointer(len)) {
-      var ptrOffset = ((len - POINTER_VALUE) << 8) + reader.readUint8();
-      var pos = reader.getOffset();
+      const ptrOffset = ((len - POINTER_VALUE) << 8) + reader.readUint8();
+      const pos = reader.getOffset();
       reader.setOffset(ptrOffset);
       labels = labels.concat(readHostname(reader));
       reader.setOffset(pos);
       break;
     } else {
-      var label = '';
-      for (var i = 0; i < len; ++i) {
-        label += String.fromCharCode(reader.readUint8());
-      }
+      let label = '';
+      for (let i = 0; i < len; ++i) label += String.fromCharCode(reader.readUint8());
```

Comment:
I'd wrap single statement blocks into curly braces anyway, in my opinion it's easier to read. Single return statement is fine without braces though.